File: project/Scripts/UI/PauseScreen.cs
using Godot;
using System;

public class PauseScreen : Control
{
    // Declare member variables here. Examples:
    // private int a = 2;
    // private string b = "text";
    AudioStreamPlayer bloop;
    // Called when the node enters the scene tree for the first time.
    public override void _Ready()
    {
        bloop = GetNode<AudioStreamPlayer>("bloop");
    }
    public override void _Process(float delta)
    {
        if (Input.IsActionJustPressed("ui_cancel"))
		{
            if (GetTree().Paused ==  true)
			{
                GD.Print("unpausing....");
                resume_button_pressed();
			}
            else 
            {
                GD.Print("pausing....");
				Input.SetMouseMode(Input.MouseMode.Visible);
				Visible = true;
				GetTree().Paused = true;
            }
		}
    }
    void button_hovered()
    {
        bloop.Play();
    }
    void exit_button_pressed()
    {
        GetTree().Paused = false;
        GetTree().ChangeScene("res://Title.tscn");
    }
    void resume_button_pressed()
    {
        Visible = false;
        Input.SetMouseMode(Input.MouseMode.Captured);
        GetTree().Paused = false;
    }
}

File: src/PauseScreen.cpp
#include "PauseScreen.h"

using namespace godot;

void PauseScreen::_register_methods() {
    register_method("_process", &PauseScreen::_process);
    register_method("_ready", &PauseScreen::_ready);
    register_method("exit_button_pressed", &PauseScreen::exit_button_pressed);
    register_method("resume_button_pressed", &PauseScreen::resume_button_pressed);
    register_method("button_hovered", &PauseScreen::button_hovered);
}

PauseScreen::PauseScreen() {
    _bloop = nullptr;
    _input = nullptr;
    _config = nullptr;
}

PauseScreen::~PauseScreen() {
    // add your cleanup here
}

void PauseScreen::_init() {

}
void PauseScreen::_ready() {
    _bloop = get_node<godot::AudioStreamPlayer>("bloop");
    _input = godot::Input::get_singleton();
    _config = get_node<Config>("/root/Config");
}
void PauseScreen::_process(const double p_delta) {
    
    if (_input->is_action_just_pressed("ui_cancel") == true)
    {
        if (get_tree()->is_paused() == true)
        {
            resume_button_pressed();
        }
        else
        {
            _input->set_mouse_mode(Input::MOUSE_MODE_VISIBLE);
            set_visible(true);
            get_tree()->set_pause(true);
        }
    }
}
void PauseScreen::button_hovered() {
    _bloop->play();
}
void PauseScreen::exit_button_pressed() {
    get_tree()->set_pause(false);
    get_tree()->change_scene("res://Title.tscn");

}
void PauseScreen::resume_button_pressed() {
    _config->save_conf();
    _input->set_mouse_mode(Input::MOUSE_MODE_CAPTURED);
    set_visible(false);
    get_tree()->set_pause(false);
}

